### PR TITLE
Update "puppeteer" for v1.2.0

### PR DESCRIPTION
**Pull request by g-plane**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/GoogleChrome/puppeteer/releases/tag/v1.2.0
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/puppeteer/index.d.ts [hunk 1]
```diff
@@ -401,6 +401,8 @@ export interface StyleTagOptions {
   path?: string;
   /** Raw CSS content to be injected into frame. */
   content?: string;
+  /** Script type. Use 'module' in order to load a Javascript ES6 module. */
+  type?: string;
```
Comment (marvinhagemeister) [inline types/puppeteer/index.d.ts]: I'm assuming this should go to `ScriptTagOptions` and not to `StyleTagOptions`? The `type` property isn't available on style-tags according to https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pageaddstyletagoptions

File: types/puppeteer/index.d.ts [hunk 2]
```diff
@@ -637,6 +644,18 @@ export interface Request {
...
+   * A `redirectChain` is a chain of requests initiated to fetch a resource.
+   *
+   * - If there are no redirects and the request was successful, the chain will be empty.
+   * - If a server responds with at least a single redirect, then the chain will contain all the requests that were redirected.
+   *
+   * `redirectChain` is shared between all the requests of the same chain.
+   *
+   * @since 1.2.0
+   */
+  redirectChain(): Promise<Request[]>;
```
Comment (marvinhagemeister) [inline types/puppeteer/index.d.ts]: The docs say that `requestChain()` returns no `Promise`. Not sure if the docs are just incorrect though.

https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#requestredirectchain

**Review by marvinhagemeister — CHANGES_REQUESTED** [review]

File: types/puppeteer/index.d.ts (hunk 1 above)
Comment (g-plane) [reply types/puppeteer/index.d.ts]: So sorry. That is my mistake.

File: types/puppeteer/index.d.ts (hunk 2 above)
Comment (g-plane) [reply types/puppeteer/index.d.ts]: I will make a change.

**Review by marvinhagemeister — APPROVED** [review]
@g-plane Awesome, thanks for your PR 👍

**Pull request merged**